protocols/noise/tests/smoke.rs

```diff
@@ -40,7 +41,8 @@ fn core_upgrade_compat() {
 #[test]
 fn xx() {
     let _ = env_logger::try_init();
-    fn prop(message: Vec<u8>) -> bool {
+    fn prop(mut messages: Vec<Message>) -> bool {
+        messages.truncate(5);
```

Comment:
I am guessing that you are `truncate`ing `messages` here to prevent huge `messages` vector, right?

Why not use `TestResult::discard` when `messages` is greater than 5. Would increase the entropy of the test as it would also run with 4, 3, 2 and 1 message.

See for comparison: https://github.com/libp2p/rust-libp2p/blob/master/protocols/gossipsub/tests/smoke.rs#L171

Reply:
> I am guessing that you are `truncate`ing `messages` here to prevent huge `messages` vector, right?

Yes.

> Why not use `TestResult::discard` when `messages` is greater than 5.

It is not so much the length that matters here, but just that multiple messages are transmitted. If generation is skewed towards the upper bound 5 it is alright. Truncating the input means less input is generated only to be discarded. That being said I am not opposed to change this to use `TestResult::discard`.

> Would increase the entropy of the test as it would also run with 4, 3, 2 and 1 message.

The test runs just fine with vectors of length 0 to 4.

Reply:
I am fine either way. In case you think this would be beneficial, add it, otherwise leave it as is with static 5.